src/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.unit.cs

```diff
@@ -549,7 +553,7 @@ public void FileSystemWatcher_Renamed()
             }
         }
 
-        [Fact]
+        [ConditionalFact(nameof(RunDangerousTests))]
         [PlatformSpecific(TestPlatforms.Linux)]  // Reads MaxUsersWatches from Linux OS files
         [OuterLoop("This test has high system resource demands and may cause failures in other concurrent tests")]
```

Comment:
I moved this to separate no-parallel class instead of making this conditional. That can still possibly break some OS functions but it should not impact other tests as they will not run at the same time.